### ISSUE TITLE
cli for sync and watch - ci servers

### ISSUE THREAD

**Issue by schmitch** [rancher/fleet#387]
hello, a good practice when it comes to k8s cd is to actually use a ci system.

currently in argo they have a good best practice to deploy an application via ci: https://argoproj.github.io/argo-cd/user-guide/ci_automation/#synchronize-the-app-optional

so basicaly a good thing in a ci system would involve:

1. update docker images
2. pull the git repo with the config
3. change the config to use the newly pushed images
4. (optional) sync the config/cd && watch it

i think watching and syncing is a good thing in the ci pipeline since it brings full ci/cd with gitlab-ci or github-actions besides that it uses a different system to keep the cluster state in sync.

basically the cli can either only work with rancher or directly with k8s (if credentials are provided) (since fleet can also work just with k8s crd's) (from an authentication perspective it would probably best to put the cli only into rancher)
